File: Chapter1/one-five.py
# one edit away

def check(one, two):
    if (len(one) == len(two)):
        return replace(one, two)
    elif(len(one) == len(two)+1):
        return single_dif(one, two)
    elif(len(one) == len(two)-1):
        return single_dif(two, one)
    else:
        return False

def replace(one, two):
    differences = 0
    for x in range(0, len(one)):
        if (one[x] != two[x]):
            differences += 1
    if (differences <= 1):
        return True
    return False

def single_dif(one, two):
    index_one = 0
    index_two = 0
    while (index_one < len(one) and index_two < len(two)):
        if(one[index_one] != two[index_two]):
            index_one += 1
            if(index_one != index_two):
                return False
        else:
            index_one += 1
            index_two += 1
    return True

File: CrackingTheCodingInterview/Chapter1/OneFive.java
// one edit away
public class OneFive 
{

    public Boolean check(String one, String two)
    {
        if (one.length() == two.length()) 
        {
            return replace(one, two);
        }
        else if(one.length() == two.length()+1)
        {
            return single(one, two);
        }
        else if(one.length() == two.length()-1)
        {
            return single(two, one);
        }
        return false;
    }
    
    public Boolean replace(String one, String two)
    {
        int differences = 0;
        for (int i = 0; i < one.length(); i++)
        {
            if (one.charAt(i) != two.charAt(i))
            {
                differences++;
            }
        }
        if(differences <= 1)
        {
            return true;
        }
        return false;
    }

    public Boolean single(String one, String two)
    {
        int index1 = 0;
        int index2 = 0;
        while (index1 < one.length() && index2 < two.length())
        {
            if (one.charAt(index1) != two.charAt(index2))
            {
                index1++;
                if(index1 != index2)
                {
                    return false;
                }
            }
            else
            {
                index1++;
                index2++;
            }
        }
        return true;
    }
}